src/sentry/rules/processing/delayed_processing.py

```diff
@@ -142,7 +142,9 @@ def get_condition_group_results(
             return None
 
         _, duration = condition_inst.intervals[unique_condition.interval]
-        comparison_interval = condition_inst.intervals[unique_condition.interval][1]
+        comparison_interval = condition_inst.intervals[
+            condition_data.get("comparisonInterval", "5m")
+        ][1]
```

Comment:
It'd be nice to include a unit test for this to make sure we don't run into the issue again in the future.

Reply:
^ this and/or a test with a rule where the comparison interval and the comparison value are different

Reply:
+1 on a test here, we need something that reproduces this bug without the fix

Reply:
ok wrote the test and confirmed it fails with the previous code but works now